Exercise_2.js

```diff
@@ -0,0 +1,272 @@
+  return name[0] === ("A")
+})
+console.log(result);
+
+
+// Print out an array of the inventors whose name contains 'n'.
+result = inventors.filter(function (name) {
+  return name.includes("n")
+});
+console.log(result);
+
+
+// Print out an array of the inventors whose name has the same letter twice in a row (e.g. nn or mm).
+
+result = inventors.filter(function (name) {
+  for (var i = 0; i < name.length; i++) {
+    if (name[i] === name[i + 1]) {
+      return name
+    }
+  }
+});
+console.log(result);
+
+// Print out an array of the numbers which are odd.
+result = numbers.filter(function (number) {
+  if ((number % 2 === 0) && (number > 0)) {
+    return number
+  }
+});
+console.log(result);
+
+
+
+// Print out an array of the numbers that have two digits.
+result = numbers.filter(function (number) {
+  if ((number > 9) && (number < 100)) {
+    return number
+  }
+});
+console.log(result);
+
+
+// Print out an array of the numbers which are prime.
+result = numbers.filter(function (number) {
+  if (number <= 1) {
+    return false
+  }
+  for (var i = 2; i <= (Math.sqrt(number)); i++) {
+    if (number % i === 0) {
+      return false;
+    }
+    return true;
+  }
+});
+console.log(result)
+
+
+/////////////////MAP/////////////////////////
+// Print out the first name of each inventor.
+result = inventors.map(function (name) {
+  return name.split(" ")[0]
+});
+console.log(result)
+
+
+// Print out the length of every inventor's full name.
+
+result = inventors.map(function (fullName) {
+  return fullName + ": " + fullName.length
+});
+console.log(result)
+// Print out initials of all inventors(e.g. A.E., I.N., ...)
+result = inventors.map(function (name) {
+  space = name.indexOf(" ");
+  return name[0] + "." + name[space + 1]
+});
+console.log(result)
+
+// Print out an array of every number multiplied by 100.
+result = numbers.map(function (number) {
+  return 100 * number
+});
+console.log(result)
+
+////////////////////find/findIndex///////////////////////
+// Find the inventor that has a middle name.
+const countSpace = (name) => {
+  var count, i;
+  for (i = 0; i < name.length; i++)
+    if (name[i] == " ") {
+      count++
+    };
+  return count;
+}
+result = inventors.find(function (name) {
+  if (countSpace(name) >= 2) {
+    return name
+  }
+});
+console.log(result);
+
+
+// Bonus: Return a new array, that only has inventors without a middle name. (Hint: think about splice, if you use findIndex. But you may also use another of the methods you've learned about above.)
+const find_index = (name) => {
+  findIndexStart = inventors.findIndex(function (name) {
+    return countSpace(name) <= 1;
+  });
+  return findIndexStart;
+}
+
+const newArray=(name)=> {
+  var array = "";
+  startArray = find_index(inventors);
+  endArray = name.length + 1;
+  array = array + inventors.slice(startArray, endArray);
+  return array;
+}
+ var i = 0;
+ while (i< inventors.length){
+  console.log(newArray(inventors[i]))
+  i++;
+ };
```

Comment:
first: the result is incorrect (Sarah E. Goode have a middle name)
second: function should use return not use console.log into function. should return the array and console.log this array